<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClockView</class>
 <widget class="QWidget" name="ClockView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>335</width>
    <height>46</height>
   </rect>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="StartStopButton" name="startStopButton">
     <property name="text">
      <string>Start</string>
     </property>
     <property name="icon">
      <iconset resource="../icons.qrc">
       <normaloff>:/icons/start.png</normaloff>:/icons/start.png</iconset>
     </property>
     <property name="shortcut">
      <string>Alt+R</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QComboBox" name="frequency">
     <item>
      <property name="text">
       <string>0.25 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>0.5 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>5 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>10 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>100 Hz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1 kHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>10 kHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Unlimited</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="singleStepButton">
     <property name="text">
      <string>Single Step</string>
     </property>
     <property name="icon">
      <iconset resource="../icons.qrc">
       <normaloff>:/icons/step.png</normaloff>:/icons/step.png</iconset>
     </property>
     <property name="shortcut">
      <string>Alt+S</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="BitsView" name="clockState" native="true">
     <property name="minimumSize">
      <size>
       <width>22</width>
       <height>22</height>
      </size>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>BitsView</class>
   <extends>QWidget</extends>
   <header>views/BitsView.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>StartStopButton</class>
   <extends>QPushButton</extends>
   <header>views/StartStopButton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>startStopButton</sender>
   <signal>clicked()</signal>
   <receiver>ClockView</receiver>
   <slot>onStartStopButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>singleStepButton</sender>
   <signal>pressed()</signal>
   <receiver>ClockView</receiver>
   <slot>onSingleStepButtonPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>singleStepButton</sender>
   <signal>released()</signal>
   <receiver>ClockView</receiver>
   <slot>onSingleStepButtonReleased()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frequency</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>ClockView</receiver>
   <slot>onFrequencyCurrentIndexChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onStartStopButtonClicked()</slot>
  <slot>onSingleStepButtonPressed()</slot>
  <slot>onSingleStepButtonReleased()</slot>
  <slot>onFrequencyCurrentIndexChanged(int)</slot>
 </slots>
</ui>
